qiskit/transpiler/target.py

```diff
@@ -227,6 +244,20 @@ def __init__(
         self.aquire_alignment = aquire_alignment
         self._non_global_basis = None
         self._non_global_strict_basis = None
+        if qubit_properties is not None:
+            if not self.num_qubits:
+                self.num_qubits = len(qubit_properties)
+            else:
+                if self.num_qubits != len(qubit_properties):
+                    raise ValueError(
+                        "The value of num_qubits specified does not match the "
+                        "length of the input qubit_properties list"
+                    )
+        self.qubit_properties = qubit_properties
+
+    def get_qubit_properties(self, qubit):
+        """Return the qubit properties for the given qubit."""
+        return self.qubit_properties[qubit]
```

Comment:
I'm a little concerned that having similar (but different)`Backend.qubit_properties(qubit or qubits)` and `Target.qubit_properties(qubit)` might confuse users. Why don't we have the same function `Target.qubit_properties(qubit or qubits)` here and renaming the Target's `qubit_properties` to `all_qubit_properties` or something?

Reply:
Thinking about this a bit more I'm not sure what value this method actually adds. Like I'm thinking what does:

```python
target.qubit_properties(qubit)
```
do that:

```python
target.qubit_properties[qubit]
```
doesn't? Would you be ok if I just deleted this and then updated `Backend.qubit_properties()` to just access the list directly?

Reply:
That could be an option, but in that case, we might have to update `target.instruction_properties(index)` as well. I thought that's the reason why we thought of`target.qubit_properties(qubit)` first.

Reply:
The difference with `instruction_properties` is that it's not properties of a fixed sequence from 0 to n, the `instruction_properties` are defined for an operation on qubits. How would you map an index from `target["cx"][(1, 3)]` (which is how you get the instruction properties today) to an index for `target.instruction_properties()`?

Reply:
OK. I understood the function `instruction_properties()` is original to InstructionProperties and it has nothing to do with the attribute `qubit_properties`.

Also I thought `all_qubit_properties` could be a better name than `qubit_properties` to represent a list of qubit properties, but I have no strong opinion on it.

Getting back to your original point (we should have `get_qubit_properties` or not), 
I like removing `get_qubit_properties` or upgrading it to take `qubits` and return a list of QubitProperties of the qubits.
I had a case such that the upgraded one would be useful in [my preparing PR](https://github.com/Qiskit/qiskit-aer/blob/72b9651550a230219c32285e086cbe99d99c02c9/qiskit/providers/aer/noise/device/models.py#L223). But I think this might be a minor use case and adding a new API later is easier than removing. So removing `get_qubit_properties` makes more sense to me.

Reply:
In that case, I'm thinking `backend.qubit_properties()` would be updated to have `@property` annotation so that we can have a common interface  `backend.qubit_properties` and `tarnget.backend.qubit_properties`.
And I'm afraid it could be an API change, that means we might need version up.
Am I correct?

Reply:
I went with just having a public attribute for the target in: https://github.com/Qiskit/qiskit-terra/pull/7736/commits/052fc62564e91250201d06cce79ecb9b454aa909 what do you think about this approach?

Reply:
As for your earlier comments I honestly forgot about the `instruction_properties()` method, I personally never use it. It's a weird use case in my opinion where you want to return a list of all instructions and then access a particular instruction's properties based on the index from that returned list. It's almost always more efficient to just traverse the target directly and access the properties during that iteration.

As for changing the backend interface to be a property instead of a method, yeah that will be a breaking change and something we'd have to wait for in `BackendV3`. Personally I'm not too worried about a difference between the target class and the backend class here. As in most cases the backend api will just be a wrapper around the target so even if it's `backend.qubit_properties(q)` vs `target.qubit_properties[q]` I'm not too worried people will be confused especially because `backend.qubit_properties()` can take a list of qubits, while for the target the user needs to use a list comprehension or a slice to get multiple qubits.

Reply:
I agree with your approach. I personally don’t want to have `backend.qubit_properties(q)` and `target.qubit_properties[q]` but I understand more that we don’t want to version up just for that.